rmf_traffic/test/unit/test_Profile.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <rmf_traffic/Profile.hpp>
+#include <rmf_traffic/DetectConflict.hpp>
+#include <rmf_traffic/geometry/Circle.hpp>
+#include <rmf_traffic/geometry/Box.hpp>
+
+#include <rmf_utils/catch.hpp>
+
+using namespace std::chrono_literals;
+
+SCENARIO("Testing Construction")
+{
+  using Profile = rmf_traffic::Profile;
+
+  WHEN("Vicinity is not passed into the constructor")
+  {
+    const auto circle_1 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(1.0);
+    const auto circle_2 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(2.0);
+
+    auto profile = Profile{circle_1};
+
+    auto& footprint = profile.footprint();
+    auto& vicinity = profile.vicinity();
+
+    CHECK((footprint->get_characteristic_length() - 1.0) == Approx(0.0).margin(1e-6));
+    CHECK((vicinity->get_characteristic_length() - 1.0) == Approx(0.0).margin(1e-6));
+
+    // Footprint and vicinity are updated
+    profile.vicinity(circle_2);
+    profile.footprint(circle_2);
```

Comment:
Maybe we should consider a test where `vicinity` gets set while `footprint` remains the same.

Reply:
I've updated the tests to include sub-blocks for 1) Only footprint is updated 2) Only vicinity is updated and 3) Both footprint and vicinity are updated.